tests/libmonitoring/metric_matcher.go

```diff
@@ -0,0 +1,70 @@
+package libmonitoring
+
+import (
+	"fmt"
+
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+	"github.com/onsi/gomega/format"
+)
+
+type MetricMatcher struct {
+	Metric operatormetrics.Metric
+	Labels map[string]string
+}
+
+func (matcher *MetricMatcher) FailureMessage(actual interface{}) (message string) {
+	msg := format.Message(actual, "to contain metric", matcher.Metric.GetOpts().Name)
+
+	if matcher.Labels != nil {
+		msg += fmt.Sprintf(" with labels %v", matcher.Labels)
+	}
+
+	return msg
+}
+
+func (matcher *MetricMatcher) NegatedFailureMessage(actual interface{}) (message string) {
+	msg := format.Message(actual, "not to contain metric", matcher.Metric.GetOpts().Name)
+
+	if matcher.Labels != nil {
+		msg += fmt.Sprintf(" with labels %v", matcher.Labels)
+	}
+
+	return msg
+}
+
+func (matcher *MetricMatcher) Match(actual interface{}) (success bool, err error) {
+	actualMetric, ok := actual.(promResult)
+	if !ok {
+		return false, fmt.Errorf("metric matcher requires a libmonitoring.PromResult")
+	}
+
+	actualName, ok := actualMetric.Metric["__name__"]
+	if !ok {
+		return false, fmt.Errorf("metric matcher requires a map with __name__ key")
+	}
+
+	nameToMatch := matcher.Metric.GetOpts().Name
+	if matcher.Metric.GetType() == operatormetrics.HistogramType || matcher.Metric.GetType() == operatormetrics.HistogramVecType {
+		nameToMatch = nameToMatch + "_bucket"
```

Comment:
Is this a standard suffix added by Prometheus libs when the type is historgram? Can we make a global const with a meaningful name?

Reply:
added